Fixes #3890: Shared link bypasses biometrics

sblatz opened the pull request:
### Demo

![new behavior](https://user-images.githubusercontent.com/4400286/48509718-d893df80-e81f-11e8-94b0-75ab50540239.gif)

ekager commented on app/src/main/java/org/mozilla/focus/biometrics/BiometricAuthenticationDialogFragment.kt:
```diff
@@ -80,11 +85,17 @@ class BiometricAuthenticationDialogFragment : AppCompatDialogFragment(), Lifecyc
     interface BiometricAuthenticationListener {
         fun onCreateNewSession()
```
Can we rename this its not clear its involving biometric and not our sessionmanager sessions

ekager commented on app/src/main/java/org/mozilla/focus/biometrics/BiometricAuthenticationDialogFragment.kt:
```diff
@@ -80,11 +85,17 @@ class BiometricAuthenticationDialogFragment : AppCompatDialogFragment(), Lifecyc
     interface BiometricAuthenticationListener {
         fun onCreateNewSession()
         fun onAuthSuccess()
+        fun onCreateNewSessionWithLink()
```
And rename this one :)

ekager commented on app/src/main/java/org/mozilla/focus/biometrics/BiometricAuthenticationDialogFragment.kt:
```diff
@@ -80,11 +85,17 @@ class BiometricAuthenticationDialogFragment : AppCompatDialogFragment(), Lifecyc
     }
 
-    fun newSessionButtonClicked() {
+    private fun newSessionButtonClicked() {
```
rename something like "biometricNewSessionButtonClicked"

colintheshots commented:
ktlint is failing:

/opt/focus-android/app/src/main/java/org/mozilla/focus/biometrics/BiometricAuthenticationHandler.kt:83:32: Parameter should be on a separate line (unless all parameters can fit a single line)
/opt/focus-android/app/src/main/java/org/mozilla/focus/biometrics/BiometricAuthenticationHandler.kt:84:32: Unexpected indentation (expected 8, actual 31)
/opt/focus-android/app/src/main/java/org/mozilla/focus/biometrics/BiometricAuthenticationHandler.kt:84:63: Missing newline before ")"

sblatz commented:
Fixed nits and added [this line](https://github.com/mozilla-mobile/focus-android/pull/3910/files#diff-9f997079c3a3095aaff772f3c980dc79R299) to account for home screen icons. Ktlint is also passing (forgot to add the hook back when my laptop got wiped). Let me know if this is safe to merge!
